hashicorp/dev-portal
# feat: HDS `Pagination`

## 🔗 Relevant links

- [Swingset  Preview][swingset_preview] 🔎
- Asana Task Attached at bottom of description 🎟️

## 🗒️ What

This adds an HDS Pagination component.
Here are some links for reference:
- (Old)
  - ~in-progress ember component: https://github.com/hashicorp/design-system/pull/661~
  - ~preview: https://hds-components-git-hds-943-pagination-component-hashicorp.vercel.app/components/pagination~
- (New)
  - PR: https://github.com/hashicorp/design-system/pull/1020
  - preview: https://hds-components-git-hds-972-pagination-cloud-ui-c491d7-hashicorp.vercel.app/components/pagination
- figma spec: https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=17236%3A219695&t=z5zFye8v2bIQmZZY-0

### TODOs

- [x] mobile styling responsibility can be handled by the consumer. Due to the _composed_ nature of this component, you can wrap `Pagination.Nav|SizeSelector|Info` in `div`s with desired classNames/styling

## 🤷 Why

To achieve design system consistency with HDS.

## 🛠️ How

N/A

## 📸 Design Screenshots

(See [Swingset  Preview][swingset_preview])

## 🧪 Testing

(See [Swingset  Preview][swingset_preview])

Things to look for
- Ensure mouse interactions work
- Ensure keyboard interactions work
- Ensure screenreader interaction works

## 💭 Anything else?

[swingset_preview]: https://dev-portal-git-kw-hds-paginator-hashicorp.vercel.app/swingset/components/pagination

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203548351777721
  - https://app.asana.com/0/0/1203653018794425
